src/crimson/osd/main_config_bootstrap_helpers.h

```diff
@@ -0,0 +1,98 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:nil -*-
+// vim: ts=8 sw=2 smarttab
+
+#pragma once
+
+#include <sys/types.h>
+#include <unistd.h>
+
+#include <iostream>
+#include <fstream>
+#include <random>
+
+#include <seastar/core/future.hh>
+
+#include "common/ceph_argparse.h"
+#include "include/expected.hpp"
+
+namespace crimson::osd {
+
+void usage(const char* prog);
+
+inline uint64_t get_nonce()
+{
+  if (auto pid = getpid(); pid == 1 || std::getenv("CEPH_USE_RANDOM_NONCE")) {
+    // we're running in a container; use a random number instead!
+    std::random_device rd;
+    std::default_random_engine rng{rd()};
+    return std::uniform_int_distribution<uint64_t>{}(rng);
+  } else {
+    return pid;
+  }
+}
+
+seastar::future<> populate_config_from_mon();
+
+struct early_config_t {
+  std::vector<std::string> early_args;
+  std::vector<std::string> ceph_args;
+
+  std::string cluster_name{"ceph"};
+  std::string conf_file_list;
+  CephInitParameters init_params{CEPH_ENTITY_TYPE_OSD};
+
+  auto to_ptr_vector(const std::vector<std::string> &in) {
```

Comment:
Draw an assumption on the life-time of `in`.

Reply:
Added comment